Automattic/wp-calypso
# State: Simplify uploadMedia and related APIs

#### Changes proposed in this Pull Request

* Remove over-complicated `serially` abstraction
* Break up `uploadMedia` into two distinctive steps (this also fixes a bug where multi-uploads wouldn't put the transient media into the store all at the same time)
* Update APIs to all be `files, site`

#### Testing instructions

* Ensure unit tests pass
* Ensure uploading site icon, podcast cover image, classic editor image and classic editor featured image all work

Part of #43661
